contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,333 @@
+it is assigned in the k8s cluster as described 
+[here](https://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/#pod-identity) and will be unique.
+It however will be possible for each pod to take multiply dns identities.
+This is detailed in the &#39;instance identity and discovery&#39; section below.
+
+For this phase of implementation, it is proposed that the statefulset need not guarantee the 
+order of creation of the pod instances across federation (_we believe more discussion will follow this statement_).
+
+### Replica distribution (across federated clusters)
+
+The proposed default behaviour of the federation controller, when a statefulset creation request is 
+sent to federation API server, would be to partition the statefulset replicas and create a statefulset 
+into each of the clusters with the reduced replica number (after partitioning), quite similar to the 
+behaviour of the replicaset or daemonset controllers of the k8s federation.
+If the replica count assigned to the federated statefulset is equal to or smaller than the total 
+healthy clusters in the federation, then there will be at most one replica assigned to the statefulset 
+created in each underlying cluster (and some clusters might not get a statefulset object at all 
+in case of replica number is smaller than total healthy clusters).
+For the case of replica numbers being larger than the total number of clusters, the proposed default 
+behaviour will be, to partition the replica count equally into healthy clusters.
+Annotations could be provided to override this behaviour in exactly the same fashion annotations 
+are provided in the spec of federated replicasets ([_frs preferences_](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/federated-replicasets.md#federatereplicaset-preferences)).
+The applications which are being deployed as the statefulset can choose to connect to the other 
+replica instances locally within the cluster or globally across the clusters to form the quorum 
+using the respective dns names.
+Applications (in case they support such a behaviour) can also communicate across quora (for example 
+a local db cluster connects to another cluster/instance outside the cluster for periodic backups) 
+using the global dns identities.
+
+### Instance identity and discovery
+
+Together with the statefulset object, the user will also need to submit the request for a headless 
+service corresponding to the statefulset.
+When such requests (one for statefulset and another for the headless service) are sent to the 
+federation control plane, the federation service controller will create a matching headless service 
+in each of the clusters.
+It will further partition the total number of replicas and create statefulsets with partitioned 
+replica numbers into at least 1 or more clusters.
+The noteworthy point is the proposal that federated stateful controller would additionally modify 
```

Comment:
Thanks for the specific suggestion, and apologies for the delayed response.
I have added as solution for this implementable as an admission controller, but given that its a difficult to hit problem, it might not be a necessary implementation in the first phase solution. Is that ok?